rpc_util.go

```diff
@@ -586,14 +586,14 @@ type parser struct {
 	// r is the underlying reader.
 	// See the comment on recvMsg for the permissible
 	// error types.
-	r io.Reader
+	r transport.Reader
```

Comment:
`transport` is an internal package, so it's not actually public. But I can move it to `rpc_util` and make it private, especially since that's the only place it's actually used.

Reply:
Should I declare it like this:
```go
// parser reads complete gRPC messages from the underlying reader.
type parser struct {
	// r is the underlying reader.
	// See the comment on recvMsg for the permissible
	// error types.
	r interface {
		Read(n int) (mem.BufferSlice, error)
	}

	// The header of a gRPC message. Find more detail at
	// https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md
	header [5]byte

	// bufferPool is the pool of shared receive buffers.
	bufferPool mem.BufferPool
}
```
Or should I actually declare a standalone type, like this?
```go
type bufferReader interface{
	Read(n int) (mem.BufferSlice, error)
}

// parser reads complete gRPC messages from the underlying reader.
type parser struct {
	// r is the underlying reader.
	// See the comment on recvMsg for the permissible
	// error types.
	r bufferReader

	// The header of a gRPC message. Find more detail at
	// https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md
	header [5]byte

	// bufferPool is the pool of shared receive buffers.
	bufferPool mem.BufferPool
}
```
There are no other uses of this interface

Reply:
> `transport` is an internal package, so it's not actually public

But `rpc_util.go` is part of the top-level `grpc` package and has access to everything inside `internal`.

Reply:
Yeah but I meant that it's not actually public to users of `grpc`, it's only public internally.